### PR TITLE
Handle max number of keys created nicer

### PR DESCRIPTION
**Pull request by bergman**
This kind of exception isn't handled properly hence if it happens users will see Google related exception as event description, which is not so friendly.

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesGCPServiceAccountSecretManager.java [hunk 1]
```diff
@@ -107,12 +108,14 @@ String ensureServiceAccountKeySecret(String workflowId,
     try {
       return serviceAccountSecretCache.get(serviceAccount, () ->
           getOrCreateSecret(workflowId, serviceAccount, epoch, secretName));
-    } catch (Exception e) {
+    } catch (ExecutionException | UncheckedExecutionException e) {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesGCPServiceAccountSecretManager.java]: Why do we need to be specific here?

**Comment by ulzha** [comment]
Missing description, and the test doesn't explain in what way exactly it becomes "nicer"...

**Comment by honnix** [comment]
I guess it will be nicer for user, otherwise `styx e ...` will show a long Google exception message which we'd better hide.

Somewhere here: https://github.com/spotify/styx/blob/7b9b0d934069afb24b24b3baf4f2a54226e5168d/styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java#L105-L105

File: styx-common/src/main/java/com/spotify/styx/util/GcpUtil.java [hunk 2]
```diff
@@ -38,8 +38,21 @@ public static boolean isPermissionDenied(GoogleJsonResponseException e) {
...
-        .map("PERMISSION_DENIED"::equals)
+    return "PERMISSION_DENIED".equals(error.get("status"));
+  }
+
+  public static boolean isResourceExhausted(Throwable t) {
+    return t instanceof GoogleJsonResponseException
+           && isResourceExhausted((GoogleJsonResponseException) t);
+  }
+
+  public static boolean isResourceExhausted(GoogleJsonResponseException e) {
```
Comment (fabriziodemaria) [inline styx-common/src/main/java/com/spotify/styx/util/GcpUtil.java]: A few of these methods can be private

**Comment by fabriziodemaria** [comment]
Smaller comment, looks good to me.

File: styx-common/src/main/java/com/spotify/styx/util/GcpUtil.java (hunk 2 above)
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/util/GcpUtil.java]: I think the idea was to have them public as help methods (_might be used later but not now_ stuff).

**Comment by honnix** [comment]
@ulzha @fabriziodemaria are we good to go?

**Pull request merged**